lib/Parse/Parser.cpp

```diff
@@ -505,6 +505,8 @@ void Parser::Initialize() {
     Ident_itype = &PP.getIdentifierTable().get("itype");
     Ident_rel_align = &PP.getIdentifierTable().get("rel_align");
     Ident_rel_align_value = &PP.getIdentifierTable().get("rel_align_value");
+    Ident_dynamic_bounds_cast = &PP.getIdentifierTable().get("dynamic_bounds_cast");
```

Comment:
This should start with an underscore and a capital letter, to avoid possible name clashes with existing names.    The spec needs to be updated to reflect this.